### PR TITLE
LPS-188818 - Keep pagination info when navigating tabs

### PR DESCRIPTION
**Pull request by juanjofgliferay**
Follows https://github.com/liferay-frontend/liferay-portal/pull/3502

### Problem
In the FDS View component the user could change information using the different tabs available. When a user changes the pagination info , save it, navigate to other tab and come back to the Pagination one, the old information is still visible.
The same problem occurs in the Details tab.

When the FDS View first render, it request the **general view** data, that is passed as a prop to the different tabs. To trigger a new data request the only dependency is the viewId, that it is something that does not change among the tabs, so whenever a user changes the tab, is loading stale **general view** data. This is also happening with the other tabs, but don't have relevant data to update the UI.

### Proposed solution
The parent FDS View passes an update function to the components that need to update the view after saving the data.
In the current PR, I'm only passing the callback to the Details and Pagination tabs, but could be passed to all of them to be consistent. 

### Before
[before.webm](https://github.com/liferay-frontend/liferay-portal/assets/132653342/415b2f81-b2bc-4a76-a9f7-f4540c96046f)

### After
[after.webm](https://github.com/liferay-frontend/liferay-portal/assets/132653342/a18ca721-df08-434a-9201-a4701776cea8)

### CODE REVIEW

**Comment by juanjofgliferay** [comment]
@dsanz @markocikos Recreating PR from scratch to avoid conflicts

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/FDSView.tsx [hunk 1]
```diff
@@ -140,6 +141,11 @@ const FDSView = ({
 						fdsView={fdsView}
 						fdsViewsURL={fdsViewsURL}
 						namespace={namespace}
+						onFDSViewUpdate={(
+							updatedViewData: FDSViewType
+						): void => {
+							setFDSView({...fdsView, ...updatedViewData});
+						}}
```
Comment (kresimir-coko) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/FDSView.tsx:148]: Do we need to specify the type here? I assume that specifying it above on line 60 makes it unnecessary to specify on line 145-146

Comment (juanjofgliferay) [reply modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/FDSView.tsx:148]: Probably not. Just inlined the function from an external declaration and forgot to remove.

**Comment by juanjofgliferay** [comment]
ci:test:relevant

**Comment by juanjofgliferay** [comment]
ci:forward

**Comment by dsanz** [comment]
@mateusralv this one is also ready to check the relevant, pls let me know :heart:

**Comment by mateusralv** [comment]
I'm taking a look 🕵🏽

**Comment by mateusralv** [comment]
ci:test:frontend-data-set

**Comment by mateusralv** [comment]
Hi @juanjofgliferay and @dsanz the relevant failures are not related.

I ran `ci:test:frontend-data-set` just to double check 😄 

[Testray Comparison](https://testray.liferay.com/home/-/testray/runs/compare?formDate=1688142938900&testrayRunIdA=2366214663&testrayRunIdB=2366423667)

**Comment by dsanz** [comment]
ci:forward:force

**Comment by mateusralv** [comment]
The failures of this PR (https://github.com/liferay-frontend/liferay-portal/pull/3512#issuecomment-1615026259) are unrelated, and safe to forward!

**Pull request closed without merging**